modules/ai/site-planner-connect/module.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Elementor\Modules\Ai\SitePlannerConnect;
+
+defined( 'ABSPATH' ) || exit;
+
+class Module {
+
+	public function __construct() {
+		add_action( 'rest_api_init', [ $this, 'on_rest_init' ] );
+		add_action( 'admin_menu', [ $this, 'register_menu_page' ], 100 );
+		add_filter('rest_prepare_application_password', function ( $response, $item, $request ) {
+			if ( $request->get_route() === '/wp/v2/users/me/application-passwords' && is_user_logged_in() ) {
+				$user = wp_get_current_user();
+				$response->data['user_login'] = $user->user_login;
+			}
+			return $response;
+		}, 10, 3);
+	}
+
+	public function on_rest_init(): void {
+		( new WpRestApi() )->register();
+	}
+
+	public function register_menu_page() {
+		add_submenu_page(
+			null, // Hidden page
```

Comment:
why this comment? A comment is a code smell. Why not have a const called `HIDDEN_PAGE` with the value of `null` instead - it's cleaner

Reply:
Comments are faster in runtime: https://gist.github.com/matipojo/f425185ad12d9cc628163120f2b99640